### ISSUE TITLE
[BUG] GUI freezes when opening a directory with a lot of pictures

### ISSUE THREAD

**Issue by mstojanovic** [owncloud/android#2774]
### Actual behavior
- when i open the directory that contains many pictures (800+ items) the dots above start showing as it is loading thumbnails. after some time the GUI just freezes and i need to force close the app. if i need to upload something to that directory i need to be fast to set the upload, then the GUI stops responding and i need to wait while the upload finishes before i can force close the app. Also if theres a lot of files on the upload queue many fail so i have to re upload from the upload menu (items get copy locally but not uploaded to the server).

also, what i think is really important for my work flow is that i dont have all the files locally, i.e. i delete them from my phone since i dont need 30+gb files on my phone after i upload them.

### Expected behaviour
- directory should open and GUI should not freeze
 
### Steps to reproduce
all described in above point

### (url: https://demo.owncloud.org, user: test, password: test)
have not tested

### Environment data
Android version: 9

Device model: samsung s9

Stock or customized system: stock

ownCloud app version: 2.14.1

ownCloud server version: 10.3? latest available im pretty sure

#### Web server error log
not needed for this issue, purely a client bug

#### ownCloud log (data/owncloud.log)
looking at log is useless due to the amount of thumbnails that need to be created (and i cant really share it unless you look for something specific, then i can maybe clean the log. protection of personal data, sorry)

**Comment by jesmrec** [comment]
Thanks for reporting. Will check which is the cause and the best solution

**Comment by jesmrec** [comment]
Trying to reproduce the problem, i did not get an ANR with a folder that contains +1000 pictures. However, i found many logs like the following one (Android 9, app 2.14):

```
2020-01-13 12:56:26.223 27043-27043/com.owncloud.android.debug V/(ThumbnailsCacheManager.java:590) .cancelPotentialThumbnailWork(): Cancelled generation of thumbnail for a reused imageView
2020-01-13 12:56:26.230 27043-27756/com.owncloud.android.debug W/System.err: java.io.InterruptedIOException: interrupted
2020-01-13 12:56:26.230 27043-27756/com.owncloud.android.debug W/System.err:     at okio.Timeout.throwIfReached(Timeout.java:146)
2020-01-13 12:56:26.230 27043-27756/com.owncloud.android.debug W/System.err:     at okio.Okio$1.write(Okio.java:76)
```

Some questions: 

- Your +800 pics, are they using 30+gb? not sure if i understood correctly.
- Are your stuff only downloaded or even set as av. offline (purple mark)?

**Comment by mstojanovic** [comment]
i have the same error after every crash.
no i have multiple folders with 800+ pictures that in total have 30+gb
only 'downloaded', i.e. green mark, but not the full directory i dont keep the full local copy of the dirs on my phone.

i did some more tests, if i open a dir it takes some time to load but i loads thumbnails, the issue is only after i set files for upload to those big dirs, after selecting files for upload and start the upload the gui freezes but the upload continues. after the upload is done (i.e. no more notifications change in the notif bar) i need to close the app and restart it. (and that notification status also has some improvement space, i often dont get upload failed errors since each file starts a new notif action. i think theres a enhancement reported to fix it already)

**Comment by DeepDiver1975** [comment]
🤖 Automated issue triage by Claude Code review agent.

Thanks for the original report and the detailed follow-up. This was filed in late 2019 against app **2.14**, and at that time @jesmrec was unable to reproduce an ANR even with a folder of 1000+ pictures — the freeze you described was tied specifically to *uploading* many files from large directories rather than just opening them.

Since then the ownCloud Android app has been substantially reworked (current release is **4.8.x**), including changes to thumbnail loading and the upload/notification pipeline, so the original code paths from 2.14 no longer apply.

Because we can't confirm this behavior on the current codebase, we're treating it as needs-info. If you (or anyone) can still reproduce a GUI freeze when opening or uploading from a large picture-heavy directory on a **current release (4.8.x)**, please **reopen** or open a fresh issue with:

- app version + Android version,
- rough folder size / file count,
- whether it happens on open vs. during upload, and ideally a log snippet around the freeze.

Closing for now as outdated — thanks again.

🤖 Generated with [Claude Code](https://claude.com/claude-code)
